src/crypto/secret.rs

```diff
@@ -23,29 +21,38 @@ impl Secret {
 
         iv
     }
+
+    #[cfg(test)]
+    pub(crate) fn from_test_vector(test_vector: &crate::test_vectors::TestVector) -> Secret {
+        Secret {
+            key: test_vector.key.clone(),
+            salt: test_vector.salt.clone(),
+            #[cfg(not(feature = "ring"))]
```

Comment:
generally it feels better to feature gate by ciphersuite/algorithm rather than which backend we use

Reply:
I agree, also the API now is not symmetric depending on the crypto backend. So either I could add a new feature `aes_ctr` which requires openssl. However I fear this could make the configuration of this library a bit complicated/nonintuitiv.

Another option would be to not feature gate `CipherSuiteVariant::{AesCtr128HmacSha256_80,
    AesCtr128HmacSha256_64,
    AesCtr128HmacSha256_32,}` at all, but rather could panic at runtime.
To notify the user with a compiler warning, one could misuse the deprecated flag for now (as long the Diagnostic API is still nightly):
```
pub enum CipherSuiteVariant {
    // /// counter mode is [not implemented in ring](https://github.com/briansmith/ring/issues/656)
    #[cfg_attr(all(feature = "ring"), deprecated = "Ring does not support, AES-Ctr")]
    AesCtr128HmacSha256_80,
    #[cfg_attr(all(feature = "ring"), deprecated = "Ring does not support, AES-Ctr")]
    AesCtr128HmacSha256_64,
    #[cfg_attr(all(feature = "ring"), deprecated = "Ring does not support, AES-Ctr")]
    AesCtr128HmacSha256_32,
    /// encryption: AES GCM 128, key expansion: HKDF with SHA256
    AesGcm128Sha256,
    /// encryption: AES GCM 256, key expansion: HKDF with SHA512
    AesGcm256Sha512,
}
```
Currently I would prefer this option.

Reply:
pffff, thats difficult, maybe we should then rather stick with the backend-vs-backend configuration instead. make a nice table and document what is supported where.
If you have the means to make something not compile at all, you should rather not make it panic at runtime. But I think it would be an improvement to remove the `not()`s from the configuration, that makes it a bit harder to reason about I felt

Reply:
Agreed, let's keep it simply for know. Maybe the diagnostic API can offer us more possiblities in the future, let's see.

Reply:
So I removed the `not()`s and replaced it. This meant however removing the fallback to `ring`. However thinking about it, I am fine with rather throwing a compile error on misconfiguration, than a silent fallback.